### ISSUE TITLE
SIGSEGV when comparing comptime-known string members of structs iterated by `inline for` with labelled break

### ISSUE THREAD

**Issue by hazeycode** [ziglang/zig#11606]
### Zig Version

0.10.0-dev.2058+a839ccc15

### Steps to Reproduce

Try to build the following program:
[Link to demo program](https://zigbin.io/9cc45e)

### Expected Behavior

Zig should compile the program or emit a compile error message

### Actual Behavior

> 'zig build-exe test.zig' terminated by signal SIGSEGV (Address boundary error)

Output from gdb:

```
Starting program: /nix/store/s8671wqmyizca1vy8qxpwkmyn530qbxr-zig-0.10.0-dev.2058+a839ccc15/bin/zig run test.zig
[New LWP 127719]
[New LWP 127720]
[New LWP 127721]
[New LWP 127722]
[New LWP 127723]
[New LWP 127724]
[New LWP 127725]
[New LWP 127726]
[New LWP 127727]
[New LWP 127728]
[New LWP 127729]
[New LWP 127730]
[New LWP 127731]
[New LWP 127732]
[New LWP 127733]
[New LWP 127734]

Thread 1 "zig" received signal SIGSEGV, Segmentation fault.
0x0000000002a29bf0 in ?? ()
(gdb) trace
Tracepoint 1 at 0x2a29bf0
```

**Comment by hazeycode** [comment]
If the if statement within the loop in the demo program is explicitly marked `comptime` the behavior changes.
Change `if (std.mem.eql(u8, thing.name, "third"))`
to `comptime if (std.mem.eql(u8, thing.name, "third"))`

output:
> ir.cpp:5425: thread 155185 panic: assertion failed. This is a bug in the Zig compiler.
Unable to dump stack trace: debug info stripped

I will followup again with more info when I have some time to build Zig in debug

**Comment by ifreund** [comment]
I'm pretty confident that the problem here is having any break at all inside an `inline for` (or `inline while`) loop. I've never seen this work, it's always crashed the compiler for me. This can however be worked around by manually implementing the semantics of break using a boolean until this stage1 bug is fixed.

**Comment by Vexu** [comment]
Fixed by 4618c41fa6ca70f06c7e65762d2f38d57b00818c
